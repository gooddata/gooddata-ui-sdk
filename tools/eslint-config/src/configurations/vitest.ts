// (C) 2025 GoodData Corporation

import type { IConfiguration } from "../types.js";

const configuration: IConfiguration<"@vitest"> = {
    packages: [
        {
            name: "@vitest/eslint-plugin",
            version: "1.3.5",
        },
    ],
    plugin: "@vitest",
    extends: ["plugin:@vitest/legacy-recommended"],
    rules: {
        "@vitest/expect-expect": "off",
        "@vitest/no-commented-out-tests": "warn",
        "@vitest/valid-title": "warn",
        "@vitest/no-disabled-tests": "warn",
        "@vitest/no-focused-tests": "warn",
        "@vitest/no-identical-title": "warn",
        "@vitest/valid-expect": "warn",
    },
};

export default configuration;
