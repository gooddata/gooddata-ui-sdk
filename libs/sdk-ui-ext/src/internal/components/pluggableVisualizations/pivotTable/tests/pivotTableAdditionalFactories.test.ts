// (C) 2022 GoodData Corporation
import {
    newAttributeColumnLocator,
    newTotalColumnLocator,
    newWidthForAllColumnsForMeasure,
    newWidthForAllMeasureColumns,
    newWidthForAttributeColumn,
    setNewWidthForSelectedColumns,
} from "@gooddata/sdk-ui-pivot";
import {
    factoryNotationForAllMeasureColumnWidthItem,
    factoryNotationForAttributeColumnLocator,
    factoryNotationForAttributeColumnWidthItem,
    factoryNotationForMeasureColumnWidthItem,
    factoryNotationForWeakMeasureColumnWidthItem,
} from "../pivotTableAdditionalFactories.js";

import { describe, it, expect } from "vitest";

describe("pivotTableAdditionalFactories", () => {
    describe("factoryNotationForAttributeColumnWidthItem", () => {
        it("should handle item without grow to fit", () => {
            const input = newWidthForAttributeColumn("some-id", 123);
            const actual = factoryNotationForAttributeColumnWidthItem(input);
            expect(actual).toMatchSnapshot();
        });
        it("should handle item with grow to fit", () => {
            const input = newWidthForAttributeColumn("some-id", 123, true);
            const actual = factoryNotationForAttributeColumnWidthItem(input);
            expect(actual).toMatchSnapshot();
        });
    });

    describe("factoryNotationForMeasureColumnWidthItem", () => {
        const attributeLocators = [newAttributeColumnLocator("attr-id")];
        const totalLocators = [newTotalColumnLocator("attr-id", "max")];

        describe.each([
            ["numeric", 123],
            ["string", "auto"],
        ] as const)("with %s width value", (_, width) => {
            it("should handle item without grow to fit", () => {
                const input = setNewWidthForSelectedColumns("some-id", attributeLocators, width);
                const actual = factoryNotationForMeasureColumnWidthItem(input);
                expect(actual).toMatchSnapshot();
            });
            it("should handle item with grow to fit", () => {
                const input = setNewWidthForSelectedColumns("some-id", attributeLocators, width, true);
                const actual = factoryNotationForMeasureColumnWidthItem(input);
                expect(actual).toMatchSnapshot();
            });
            it("should handle item with transposed metrics and without grow to fit", () => {
                const input = setNewWidthForSelectedColumns(
                    ["measure-1", "measure-2", "measure-3"],
                    [...attributeLocators, ...totalLocators],
                    width,
                    true,
                );
                const actual = factoryNotationForMeasureColumnWidthItem(input);
                expect(actual).toMatchSnapshot();
            });
            it("should handle item with transposed metrics and with grow to fit", () => {
                const input = setNewWidthForSelectedColumns(
                    ["measure-1", "measure-2"],
                    [...attributeLocators, ...totalLocators],
                    width,
                    true,
                );
                const actual = factoryNotationForMeasureColumnWidthItem(input);
                expect(actual).toMatchSnapshot();
            });
        });
    });

    describe("factoryNotationForAttributeColumnLocator", () => {
        it("should handle item without element", () => {
            const input = newAttributeColumnLocator("some-id");
            const actual = factoryNotationForAttributeColumnLocator(input);
            expect(actual).toMatchSnapshot();
        });
        it("should handle item with element", () => {
            const input = newAttributeColumnLocator("some-id", "some-element");
            const actual = factoryNotationForAttributeColumnLocator(input);
            expect(actual).toMatchSnapshot();
        });
    });

    describe("factoryNotationForWeakMeasureColumnWidthItem", () => {
        it("should handle item without grow to fit", () => {
            const input = newWidthForAllColumnsForMeasure("some-id", 123);
            const actual = factoryNotationForWeakMeasureColumnWidthItem(input);
            expect(actual).toMatchSnapshot();
        });
        it("should handle item with grow to fit", () => {
            const input = newWidthForAllColumnsForMeasure("some-id", 123, true);
            const actual = factoryNotationForWeakMeasureColumnWidthItem(input);
            expect(actual).toMatchSnapshot();
        });
    });

    describe("factoryNotationForAllMeasureColumnWidthItem", () => {
        it("should handle item without grow to fit", () => {
            const input = newWidthForAllMeasureColumns(123);
            const actual = factoryNotationForAllMeasureColumnWidthItem(input);
            expect(actual).toMatchSnapshot();
        });
        it("should handle item with grow to fit", () => {
            const input = newWidthForAllMeasureColumns(123, true);
            const actual = factoryNotationForAllMeasureColumnWidthItem(input);
            expect(actual).toMatchSnapshot();
        });
    });
});
