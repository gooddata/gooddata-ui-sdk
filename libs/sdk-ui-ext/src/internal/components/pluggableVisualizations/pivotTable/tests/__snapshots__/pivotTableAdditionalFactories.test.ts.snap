// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pivotTableAdditionalFactories > factoryNotationForAllMeasureColumnWidthItem > should handle item with grow to fit 1`] = `"newWidthForAllMeasureColumns(123, true)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForAllMeasureColumnWidthItem > should handle item without grow to fit 1`] = `"newWidthForAllMeasureColumns(123)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForAttributeColumnLocator > should handle item with element 1`] = `"newAttributeColumnLocator("some-id", "some-element")"`;

exports[`pivotTableAdditionalFactories > factoryNotationForAttributeColumnLocator > should handle item without element 1`] = `"newAttributeColumnLocator("some-id")"`;

exports[`pivotTableAdditionalFactories > factoryNotationForAttributeColumnWidthItem > should handle item with grow to fit 1`] = `"newWidthForAttributeColumn("some-id", 123, true)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForAttributeColumnWidthItem > should handle item without grow to fit 1`] = `"newWidthForAttributeColumn("some-id", 123)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with numeric width value > should handle item with grow to fit 1`] = `"setNewWidthForSelectedColumns(["some-id"], [newAttributeColumnLocator("attr-id")], 123, true)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with numeric width value > should handle item with transposed metrics and with grow to fit 1`] = `"setNewWidthForSelectedColumns(["measure-1","measure-2"], [newAttributeColumnLocator("attr-id"), newTotalColumnLocator("attr-id", "max")], 123, true)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with numeric width value > should handle item with transposed metrics and without grow to fit 1`] = `"setNewWidthForSelectedColumns(["measure-1","measure-2","measure-3"], [newAttributeColumnLocator("attr-id"), newTotalColumnLocator("attr-id", "max")], 123, true)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with numeric width value > should handle item without grow to fit 1`] = `"setNewWidthForSelectedColumns(["some-id"], [newAttributeColumnLocator("attr-id")], 123)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with string width value > should handle item with grow to fit 1`] = `"setNewWidthForSelectedColumns(["some-id"], [newAttributeColumnLocator("attr-id")], "auto")"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with string width value > should handle item with transposed metrics and with grow to fit 1`] = `"setNewWidthForSelectedColumns(["measure-1","measure-2"], [newAttributeColumnLocator("attr-id"), newTotalColumnLocator("attr-id", "max")], "auto")"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with string width value > should handle item with transposed metrics and without grow to fit 1`] = `"setNewWidthForSelectedColumns(["measure-1","measure-2","measure-3"], [newAttributeColumnLocator("attr-id"), newTotalColumnLocator("attr-id", "max")], "auto")"`;

exports[`pivotTableAdditionalFactories > factoryNotationForMeasureColumnWidthItem > with string width value > should handle item without grow to fit 1`] = `"setNewWidthForSelectedColumns(["some-id"], [newAttributeColumnLocator("attr-id")], "auto")"`;

exports[`pivotTableAdditionalFactories > factoryNotationForWeakMeasureColumnWidthItem > should handle item with grow to fit 1`] = `"newWidthForAllColumnsForMeasure("some-id", 123, true)"`;

exports[`pivotTableAdditionalFactories > factoryNotationForWeakMeasureColumnWidthItem > should handle item without grow to fit 1`] = `"newWidthForAllColumnsForMeasure("some-id", 123)"`;
